knimin/lib/tests/test_squash_barcodes.py

```diff
@@ -1,28 +1,68 @@
-
 from unittest import TestCase, main
+from os.path import join, dirname, realpath
+from os import mkdir, path
+import random
 
-from knimin.lib.squash_barcodes import get_image, build_barcodes_pdf
+from knimin.lib.code128 import code128_image
+import knimin.lib.squash_barcodes as m
 
 
 class SquashBarcodesTests(TestCase):
+    def test_build_barcodes_pdf(self):
+        self.assertIn('Length 10559', m.build_barcodes_pdf(['000000011']))
+
+        old_get_image = m.get_image
+
+        # replace get_image method to change behaviour
+        def mock_get_image1(barcodes):
+            font = join(dirname(realpath(__file__)), '../', 'FreeSans.ttf')
+            for b in barcodes:
+                yield code128_image(b, height=150, width=300, font=font,
+                                    thickness=2, show_text=True,
+                                    quiet_zone=False)
+        m.get_image = mock_get_image1
+        self.assertIn('%PDF-1.4\n%', m.build_barcodes_pdf(['000000011']))
+
+        # replace get_image method to change behaviour
+        def mock_get_image2(barcodes):
+            font = join(dirname(realpath(__file__)), '../', 'FreeSans.ttf')
+            for b in barcodes:
+                yield code128_image(b, height=150, width=303, font=font,
+                                    thickness=2, show_text=True,
+                                    quiet_zone=False)
+        m.get_image = mock_get_image2
+        self.assertRaisesRegexp(AttributeError,
+                                "shape",
+                                m.build_barcodes_pdf,
+                                ['000000011'])
+        # reset function
+        m.get_image = old_get_image
+
+    def test_tmpdir(self):
+        # test that a temporary directory cannot collide with an existing one
+        random.seed(7)
+        fake_tmp_dir = 'tmp_IDQBN'
+        if not path.exists(fake_tmp_dir):
+            mkdir(fake_tmp_dir)
+        self.assertTrue(m.build_barcodes_pdf(['000000011']))
```

Comment:
it doesn't look like this test has anything to do with the `tmp_dir`? I'm a bit confused

On a related note, the `tempfile` provides excellent functionality for generating temp files or directories with mechanisms to ensure they're cleaned up if desired